File: finance-app-backend/src/main/java/com/victor/financeapp/backend/application/usecase/GetInvoiceUseCase.java
package com.victor.financeapp.backend.application.usecase;

import com.victor.financeapp.backend.application.dto.InvoiceDTO;
import reactor.core.publisher.Mono;

import java.time.YearMonth;

public interface GetInvoiceUseCase {

    Mono<InvoiceDTO> execute(Long creditCardId, YearMonth yearMonth);
}

File: finance-app-android/app/src/main/java/com/viictrp/financeapp/domain/usecase/GetInvoiceUseCase.kt
package com.viictrp.financeapp.domain.usecase

import com.viictrp.financeapp.data.remote.dto.InvoiceDTO
import com.viictrp.financeapp.domain.repository.BalanceRepository
import java.time.YearMonth
import javax.inject.Inject

class GetInvoiceUseCase @Inject constructor(
    private val repository: BalanceRepository
) {
    suspend operator fun invoke(creditCardId: Long, yearMonth: YearMonth): InvoiceDTO? {
        return repository.getInvoice(creditCardId, yearMonth)
    }
}
